### ISSUE TITLE
EndOfStreamException when click info button

### ISSUE THREAD

**Issue by dhlrunner** [Igor55x/UAAE#51]
![image](https://user-images.githubusercontent.com/37791084/146379657-87e2df1f-aafe-411b-a8b2-08577e0056f2.png)
I am using the Second Release version.

The assets bundle seems to load successfully, but when I click the info button I get an EndOfStream error. 

I need to re-export the modified assetsbundle, but the current version of the code is write function is disabled. that's why i using Second Release binary..

here is the assets file which is creating error.
[gal_uvmovie_1001_001_tex_00.zip](https://github.com/Igor55x/UAAE/files/7727611/gal_uvmovie_1001_001_tex_00.zip)

and here is the full exception message

> System.IO.EndOfStreamException: Unable to read beyond the end of the stream.
>    at System.IO.BinaryReader.ReadByte()
>    at AssetsTools.NET.AssetTypeTemplateField.ReadType(AssetsFileReader reader, AssetTypeValueField valueField)
>    at AssetsTools.NET.AssetTypeTemplateField.ReadType(AssetsFileReader reader, AssetTypeValueField valueField)
>    at AssetsTools.NET.AssetTypeTemplateField.ReadType(AssetsFileReader reader, AssetTypeValueField valueField)
>    at AssetsTools.NET.AssetTypeInstance..ctor(AssetTypeTemplateField[] baseFields, AssetsFileReader reader, Int64 filePos)
>    at AssetsTools.NET.AssetTypeInstance..ctor(AssetTypeTemplateField baseField, AssetsFileReader reader, Int64 filePos)
>    at AssetsAdvancedEditor.Assets.AssetsWorkspace.MakeAssetContainer(AssetItem item, Boolean forceFromCldb) in C:\Users\Lenovo\source\repos\AssetsAdvancedEditor\AssetsAdvancedEditor\Assets\AssetsWorkspace.cs:line 130
>    at AssetsAdvancedEditor.Winforms.AssetsViewer.AddAssetItem(AssetsFileInstance fileInst, AssetFileInfoEx info, Int32 fileId) in C:\Users\Lenovo\source\repos\AssetsAdvancedEditor\AssetsAdvancedEditor\Winforms\AssetsViewer.cs:line 156
>    at AssetsAdvancedEditor.Winforms.AssetsViewer.LoadAssetsToList() in C:\Users\Lenovo\source\repos\AssetsAdvancedEditor\AssetsAdvancedEditor\Winforms\AssetsViewer.cs:line 74
>    at AssetsAdvancedEditor.Winforms.AssetsViewer..ctor(AssetsManager am, AssetsFileInstance instance, Boolean fromBundle) in C:\Users\Lenovo\source\repos\AssetsAdvancedEditor\AssetsAdvancedEditor\Winforms\AssetsViewer.cs:line 63
>    at AssetsAdvancedEditor.Winforms.Menu.btnInfo_Click(Object sender, EventArgs e) in C:\Users\Lenovo\source\repos\AssetsAdvancedEditor\AssetsAdvancedEditor\Winforms\Menu.cs:line 300
>    at System.Windows.Forms.Control.OnClick(EventArgs e)
>    at System.Windows.Forms.Button.OnClick(EventArgs e)
>    at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
>    at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
>    at System.Windows.Forms.Control.WndProc(Message& m)
>    at System.Windows.Forms.ButtonBase.WndProc(Message& m)
>    at System.Windows.Forms.Button.WndProc(Message& m)
>    at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
>    at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WM msg, IntPtr wparam, IntPtr lparam)

**Comment by Igor55x** [comment]
[Here](https://github.com/Igor55x/UAAE/files/7730695/UAAE-temporary.solution.zip) you can try a temporary solution until I release a fix.

**Comment by dhlrunner** [comment]
Thanks a lot to fixing a bug, but another problem is here,

![image](https://user-images.githubusercontent.com/37791084/146470663-838348c4-9d7a-41b6-bf8c-33d1d1395b28.png)

It successfully enters the info screen. However, when I click the Edit Texture or Export To Png/Tga option, I get a NullReferenceException error. The asset file is the same file as above.

here is a full error message
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Texture.Options.EditTextureOption.ExecutePlugin(IWin32Window owner, AssetsWorkspace workspace, List`1 selectedItems) in C:\Users\Lenovo\Desktop\UAAE-88120287e86b29f03904374a930c1575abf0c370\Plugins\Texture\Options\EditTextureOption.cs:line 35
   at AssetsAdvancedEditor.Winforms.EditDialog.btnOK_Click(Object sender, EventArgs e) in C:\Users\Lenovo\Desktop\UAAE-88120287e86b29f03904374a930c1575abf0c370\AssetsAdvancedEditor\Winforms\EditDialog.cs:line 33
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WM msg, IntPtr wparam, IntPtr lparam)
```

**Comment by Igor55x** [comment]
[Texture-fix.zip](https://github.com/Igor55x/UAAE/files/7736908/Texture-fix.zip).
Just replace the "Texture.dll" file in "Plugins" folder with one from this archive.

**Comment by dhlrunner** [comment]
Thanks for your help, but it isn't fixed (error was changed?)
The asset file is the same file as above.

here is a full error message

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Texture.Options.ExportTextureOption.SingleExport(IWin32Window owner, AssetsWorkspace workspace, AssetItem selectedItem) in C:\Users\Lenovo\Desktop\UAAE-88120287e86b29f03904374a930c1575abf0c370\Plugins\Texture\Options\ExportTextureOption.cs:line 105
   at Texture.Options.ExportTextureOption.ExecutePlugin(IWin32Window owner, AssetsWorkspace workspace, List`1 selectedItems) in C:\Users\Lenovo\Desktop\UAAE-88120287e86b29f03904374a930c1575abf0c370\Plugins\Texture\Options\ExportTextureOption.cs:line 33
   at AssetsAdvancedEditor.Winforms.EditDialog.btnOK_Click(Object sender, EventArgs e) in C:\Users\Lenovo\Desktop\UAAE-88120287e86b29f03904374a930c1575abf0c370\AssetsAdvancedEditor\Winforms\EditDialog.cs:line 33
   at System.Windows.Forms.Control.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnClick(EventArgs e)
   at System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   at System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.ButtonBase.WndProc(Message& m)
   at System.Windows.Forms.Button.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WM msg, IntPtr wparam, IntPtr lparam)
```

**Comment by Igor55x** [comment]
fixed in 8b271f5
